e2e/cluster.go

```diff
@@ -198,6 +201,24 @@ func getClusterSubnetID(ctx context.Context, cloud *azureClient, location, mcRes
 	return "", fmt.Errorf("failed to find aks vnet")
 }
 
+func getClusterVnetName(ctx context.Context, cloud *azureClient, mcResourceGroupName string) (string, error) {
+	pager := cloud.vnetClient.NewListPager(mcResourceGroupName, nil)
+
+	for pager.More() {
+		nextResult, err := pager.NextPage(ctx)
+		if err != nil {
+			return "", fmt.Errorf("failed to advance page: %w", err)
+		}
+		for _, v := range nextResult.Value {
+			if v == nil {
+				return "", fmt.Errorf("aks vnet id was empty")
```

Comment:
seems like a lot of code duplication with `getClusterSubnetID`, can we maybe refactor towards returning a common struct which contains both vnet name and subnet ID since they're both based on the same vnet?